### PR TITLE
fix: user.js: Expose extra fields in WHOWAS

### LINKED ISSUE #156: Expose extra data from WHOWAS
https://github.com/kiwiirc/irc-framework/blob/622c3d7df5526aa4be487e4f6bcdf6d7ef94a5d1/src/commands/handlers/user.js#L283-L290

WHOWAS contains much more useful information: Account name, WHOISACTUALLY, and the date they were seen for instance. We should change it to be on ENDOFWHOWAS

the extra fields (assume they are nullable) are "server", "server_info"?, "account", "actualip", "actualhost", and maybe "date".)

### PR DESCRIPTION
**Pull request by Zarthus**
Closes #156 

This feature is (as of currently) untested. I will perform some testing before I consider it stable. That said, I do believe the feature implementation is complete.

For description of the issue, please refer to the issue.

### CODE REVIEW

File: src/commands/handlers/user.js [hunk 1]
```diff
@@ -281,16 +281,35 @@ var handlers = {
...
-    RPL_ENDOFWHOWAS: function() {
-        // noop
+    RPL_ENDOFWHOWAS: function(command) {
+        // Because the WHOIS and WHOWAS numerics clash with eachother,
+        // a cache key will have more than what is just in RPL_WHOWASUSER.
+        // This is why we borrow from the whois.* cache key ID.
+        //
+        // This exposes some fields (that may or may not be set).
+        // Valid keys that should always be set: nick, ident, host, real_name
+        // Valid optional keys: actualip, actualhost, account, server, server_info
```
Comment (Zarthus) [inline src/commands/handlers/user.js]: The expectation is that many of those are still often set (e.g. I expect server, server_info) to be set 100% of the time, but with ircd compatibility and such it's hard to say if there's not a rare outlier and it's safer to assume that most of these really are nullables.

**Comment by prawnsalad** [comment]
Is #159 supposed to supersede this PR?

**Comment by Zarthus** [comment]
```
) <Zarthus> https://github.com/kiwiirc/irc-framework/pull/157#issuecomment-398187117
(22:59:03) <Zarthus> I branched off of that branch because it depends on it
(22:59:11) <Zarthus> (under the assumption #157 would be merged)
(22:59:26) <Zarthus> so "technically yes", I suppose.
(23:00:08) <Zarthus> #157 is able to be merged independent from #159, but if you merge #159 you also merge #157
```

**Comment by prawnsalad** [comment]
This was merged in via #157

**Pull request closed without merging**